docs/guides/testing-dbus.rst

```diff
@@ -0,0 +1,194 @@
+instantiate the :class:`sssd_test_framework.utils.sbus.DBUSDestination` class,
+providing the MultihostHost where the D-Bus services are running, the destination
+(the application) to contact and the bus to connect to.
+
+The bus can be the bus' path as a string, or one of the predefined buses:
+
+* :attr:`sssd_test_framework.utils.sbus.DBUSKnownBus.SYSTEM`
+* :attr:`sssd_test_framework.utils.sbus.DBUSKnownBus.SESSION`
+* :attr:`sssd_test_framework.utils.sbus.DBUSKnownBus.MONITOR`
+
+.. code-block:: python
+    :caption: Accessing the monitor
+
+    @pytest.mark.topology(KnownTopologyGroup.AnyProvider)
+    def test_example__monitor(client: Client):
+        client.sssd.start()
+
+        monitor = DBUSDestination(client.host, dest="sssd.monitor", bus=DBUSKnownBus.MONITOR)
+
+        sssd = monitor.getObject(objpath="/sssd")
+
+        res = sssd.debug_level
+        assert res == 0xFFF0
+
+        sssd.debug_level = 0x0070
+
+
+A Note On Type Conversion
+~~~~~~~~~~~~~~~~~~~~~~~~~
+All methods and properties accept and return Python types. Internally they are
+converted to some specific classes helping to treat them and map them to D-Bus
+types.
+
+Objects of these types represent values that are passed to/from the methods and
+properties. Their type is given by the class. For instance, DBUSTypeString is a
+D-Bus string.
+
+These classes are all subclasses of the abstract class
+:class:`sssd_test_framework.utils.dbus.types.DBUSType` and they provide the
+following methods:
+
+* :attr:`sssd_test_framework.utils.dbus.types.DBUSType.value`: A property to read
+  and set the (Python) value.
+* :meth:`sssd_test_framework.utils.dbus.types.DBUSType.mimic()`: a method to copy
+  itself without copying the value while maintaining the structure (subtypes).
+* :meth:`sssd_test_framework.utils.dbus.types.DBUSType.param()`: The string
+  representation of the value in a format suitable to be used as a parameter for
+  ``dbus-send``.
+* :meth:`sssd_test_framework.utils.dbus.types.DBUSType.parse()`: Parses the
+  string resulting from an execution of `dbus-send` and set the value to the
+  object.
+
+Basic types (integers, strings, booleans, etc.) are subclasses of the
+:class:`sssd_test_framework.utils.dbus.types.DBUSTypeBasic` abstract class.
+
+Container types -- that is, the subclasses of the abstract class
+:class:`sssd_test_framework.utils.dbus.types.DBUSTypeContainer` -- accept a
+parameter for their constructors, another
+:class:`sssd_test_framework.utils.dbus.types.DBUSType` object of the expected type.
+
+.. code-block:: python
+    :caption: Declaring an array of unit32
+
+    s = "array [ uint32 1 uint32 2 uint32 3 ]"
+    a = DBUSTypeArray(DBUSTypeUInt32())
+    a.parse(DBUSResult(s))
+    print(a.value)
+    [1, 2, 3]
+
+In some cases it may not be possible to know in advance the type of the elements
+of a container type. In that case, no object is passed to the constructor. The
+type will be guessed while parsing the result from ```dbus-send``.
```

Comment:
remov the third backtick

Reply:
What? This is one of the things I fixed yesterday...

Checking the history, something went wrong and removed that change. I don't understand what happened.

I'll fix it again...